python/cudf/cudf/tests/test_struct.py

```diff
@@ -98,3 +98,18 @@ def test_serialize_struct_dtype(fields):
 def test_struct_getitem(series, expected):
     sr = cudf.Series(series)
     assert sr[0] == expected
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": 1, "b": "rapids", "c": [1, 2, 3, 4]},
+        {"a": 1, "b": "rapids", "c": [1, 2, 3, 4], "d": cudf.NA},
+        {"a": "Hello"},
+        {"b": [], "c": [1, 2, 3]},
```

Comment:
We also want to make sure we can construct an `NA` scalar here so long as we pass a valid `StructDtype`.